packages/netlify-cms-backend-github/src/API.js

```diff
@@ -113,13 +117,10 @@ export default class API {
   }
 
   async requestAllPages(url, options = {}) {
```

Comment:
This method didn't work for private repos as it didn't pass in the authorization header.
Also it didn't work at all under the cypress electron browser since using async iterators requires some babel config changes:

https://github.com/meteor/meteor/issues/9897
https://github.com/babel/babel/issues/7467

A `TypeError("Object is not async iterable")` was throw and `Symbol.asyncIterator` is undefined when using the cypress electron browser.
I'm assuming the same will happen with any older browser.

I re-wrote it no to use async iterators and passed in the headers to support private repos.